Cabal/Distribution/Simple/SrcDist.hs

```diff
@@ -216,7 +216,7 @@ listPackageSourcesOrdinary verbosity pkg_descr pps =
   , fmap concat
     . withAllLib $ \ l -> do
        let lbi = libBuildInfo l
-           relincdirs = "." : filter isRelative (includeDirs lbi)
+           relincdirs = "." : filter isRelativeOnAnyPlatform (includeDirs lbi)
```

Comment:
@hvr and I discussed this briefly in #hackage. We were optimistic about that `cabal sdist` would be able to run on a package not written for your platform, but maybe it's not the case.

I can change this shortly.